pkg/engine/resolve/inputtemplate.go

```diff
@@ -32,11 +33,14 @@ type InputTemplate struct {
 	// This is the case, e.g. when using batching and one sibling is null, resulting in a null value for one batch item
 	// Returning null in this case tells the batch implementation to skip this item
 	SetTemplateOutputToNullOnVariableNull bool
+	UndefinedVariables                    []string
 }
 
 var setTemplateOutputNull = errors.New("set to null")
 
 func (i *InputTemplate) Render(ctx *Context, data []byte, preparedInput *fastbuffer.FastBuffer) (err error) {
+	i.UndefinedVariables = make([]string, 0)
```

Comment:
This InputTemplate is used concurrently. It's not possible to allocate a slice on it.
Can you move UndefinedVariables to the function scope?
Or outside of the function.
But I think it cannot stay on the "shared" InputTemplate.